ui/apps/platform/src/Containers/Vulnerabilities/PlatformCves/Overview/ClustersTable.tsx

```diff
@@ -1,48 +1,145 @@
 import React from 'react';
-import { Bullseye, Spinner } from '@patternfly/react-core';
+import { pluralize } from '@patternfly/react-core';
+import { Table, Thead, Tr, Th, Td, Tbody } from '@patternfly/react-table';
+import { Link } from 'react-router-dom';
+import { gql, useQuery } from '@apollo/client';
+import noop from 'lodash/noop';
 
-import TableErrorComponent from 'Components/PatternFly/TableErrorComponent';
 import useURLPagination from 'hooks/useURLPagination';
+import { ClusterType } from 'types/cluster.proto';
+import {
+    TbodyLoading,
+    TbodyError,
+    TbodyEmpty,
+    TbodyFilteredEmpty,
+} from 'Components/TableStateTemplates';
+import { getTableUIState } from 'utils/getTableUIState';
+import { ensureExhaustive } from 'utils/type.utils';
 
+import { getPlatformEntityPagePath, getRegexScopedQueryString } from '../../utils/searchUtils';
 import { QuerySearchFilter } from '../../types';
 
+function displayClusterType(type: ClusterType): string {
+    switch (type) {
+        case 'GENERIC_CLUSTER':
+            return 'Generic';
+        case 'KUBERNETES_CLUSTER':
+            return 'Kubernetes';
+        case 'OPENSHIFT_CLUSTER':
+        case 'OPENSHIFT4_CLUSTER':
+            return 'OCP';
+        default:
+            return ensureExhaustive(type);
+    }
+}
+
+const clusterListQuery = gql`
+    query getPlatformClusters($query: String, $pagination: Pagination) {
+        clusters(query: $query, pagination: $pagination) {
+            id
+            name
+            clusterVulnerabilityCount(query: $query)
+            type
+            status {
+                orchestratorMetadata {
+                    version
+                }
+            }
+        }
+    }
+`;
+
+export type Cluster = {
+    id: string;
+    name: string;
+    clusterVulnerabilityCount: number;
+    type: ClusterType;
+    status?: {
+        orchestratorMetadata?: {
+            version: string;
```

Comment:
Still pending a response whether or not this is nullable - the change in this PR handles that case already though due to the optional chaining.

Reply:
This is definitely nullable at runtime, because some clusters do not expose this data, due to security hardening.

Reply:
Does it actually return `null`, or just `''`? It would be disappointing if it actually returned `null`/`undefined` since this would be the first case I've found where the GraphQL API contract is wrong at runtime. :(

Reply:
You may be right about it returning empty string, but paranoid me would not trust that the GraphQL API is actually runtime-typed-checked against the Go times. Could be more of a TypeScript situation.

Trust, but verify